editor/src/app/shop/shop.state.ts

```diff
@@ -21,21 +47,108 @@ export class ShopState implements NgxsOnInit {
     return state.sections;
   }
 
+  @Selector()
+  static getURLs(state: ShopModel) {
+    return state.urls;
+  }
+
   constructor(
+    private store$: Store,
     private stateService: ShopStateService) {
   }
 
 
-  ngxsOnInit({ patchState }: StateContext<ShopModel>) {
-    return this.stateService.getInitialState().pipe(
-      take(1)
+  ngxsOnInit({ patchState, setState, dispatch }: StateContext<ShopModel>) {
+    this.store$.select(SitesState).pipe(
+      filter(sites => !!sites && sites.length > 0),
+      map(sites => sites.map(site => site.name)),
+      pairwise(),
+      map(([prevSiteNames,
+        siteNames]) => {
+        let leftOverPrevSiteNames = prevSiteNames;
+        const newSiteName = siteNames.reduce((_newSiteName, siteName) => {
+          if (leftOverPrevSiteNames.indexOf(siteName) === -1) {
+            return siteName;
+          }
+          leftOverPrevSiteNames = leftOverPrevSiteNames.filter(sn => sn !== siteName);
+          return _newSiteName;
+        }, null);
+
+        return [leftOverPrevSiteNames[0], newSiteName];
+      }),
+      filter((oldSiteName, newSiteName) => !!oldSiteName || !!newSiteName)
+    ).subscribe(([oldSiteName, newSiteName]) => {
+      /**
+       * @note: Orders and Products don't need to be initialized, because there will be no orders or products
```

Comment:
Products and Order also should be initialised when new site is created by cloning other with products and orders inside.